### DISCUSSION TITLE
Retirement of Basic Microsoft SMTP auth?

### DISCUSSION THREAD

**Discussion by detroitprincess** [PHPMailer/PHPMailer#2756] [General]
With the upcoming deprecation of Microsoft's basic authentication, how would everyone suggest transitioning over to a different authentication with a valid Microsoft email account?  I assume that I'm using basic.

https://docs.microsoft.com/en-us/exchange/clients-and-mobile-in-exchange-online/deprecation-of-basic-authentication-exchange-online

**Code example:**
<?php
use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\Exception;

require 'PHPMailer/src/Exception.php';
require 'PHPMailer/src/PHPMailer.php';
require 'PHPMailer/src/SMTP.php';

$mail = new PHPMailer(true);
$mail->isSMTP();
$mail->Host = 'smtp.office365.com';
$mail->Port       = 587;
$mail->SMTPSecure = 'tls';

$mail->SMTPAuth   = true;
$mail->Username = 'myusername@mydomain.com';
$mail->Password = 'xxxxxxx';

**Comment by Synchro** [comment]
Your options are much the same as for gmail – use XOAUTH2 or app passwords. PHPMailer already has support for XOAUTH2 with MS services.
